src/Core/Checkout/Document/SalesChannel/DocumentRoute.php

```diff
@@ -152,4 +179,40 @@ private function checkGuestAuth(
             throw DocumentException::guestNotAuthenticated();
         }
     }
+
+    /**
+     * @return array<string>
+     */
+    private function getRequestedFileTypes(Request $request): array
```

Comment:
instead of returning every accepted file type, we should check until we found the first one, we are supporting. otherwise you start the generator with every kind of filetypes until one returns a valid document. that does not sound very nice for performance